### PR TITLE
Introduce ItemsResponse

### PR DESCRIPTION
**Pull request by sebm253**

### CODE REVIEW

File: rest/items_response.go [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import "github.com/disgoorg/json"
+
+// ItemsResponse serves as a utility to flatten objects wrapped in an "items" object into a slice of objects.
+type ItemsResponse[T any] []T
+
+func (rs *ItemsResponse[T]) UnmarshalJSON(data []byte) error {
+	var response struct {
+		Items []T `json:"items"`
+	}
```
Comment (sebm253) [inline rest/items_response.go:11]: would moving this into its own type make more sense?

File: rest/soundboard_sounds.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ func NewSoundboardSounds(client Client) SoundboardSounds {
 
 type SoundboardSounds interface {
 	GetSoundboardDefaultSounds(opts ...RequestOpt) ([]discord.SoundboardSound, error)
-	GetGuildSoundboardSounds(guildID snowflake.ID, opts ...RequestOpt) ([]discord.SoundboardSound, error)
+	GetGuildSoundboardSounds(guildID snowflake.ID, opts ...RequestOpt) (ItemsResponse[discord.SoundboardSound], error)
```
Comment (topi314) [inline rest/soundboard_sounds.go:16]: why break this?

File: rest/applications.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ type Applications interface {
 	DeleteTestEntitlement(applicationID snowflake.ID, entitlementID snowflake.ID, opts ...RequestOpt) error
 	ConsumeEntitlement(applicationID snowflake.ID, entitlementID snowflake.ID, opts ...RequestOpt) error
 
-	GetApplicationEmojis(applicationID snowflake.ID, opts ...RequestOpt) ([]discord.Emoji, error)
+	GetApplicationEmojis(applicationID snowflake.ID, opts ...RequestOpt) (ItemsResponse[discord.Emoji], error)
```
Comment (topi314) [inline rest/applications.go:45]: why break this?

**Review by topi314 — CHANGES_REQUESTED** [review]
I don't think `ItemResponse` should be exposed to the outside,
this type should be unexported imo

File: rest/soundboard_sounds.go (hunk 2 above)
Comment (sebm253) [reply rest/soundboard_sounds.go:16]: doesn't break anything and we need the unmarshaller

File: rest/applications.go (hunk 3 above)
Comment (sebm253) [reply rest/applications.go:45]: https://github.com/disgoorg/disgo/pull/404#discussion_r1865044552

File: rest/soundboard_sounds.go (hunk 2 above)
Comment (topi314) [reply rest/soundboard_sounds.go:16]: you changed the return type, this is breaking

Comment (sebm253) [reply rest/soundboard_sounds.go:16]: it's still a slice

Comment (topi314) [reply rest/soundboard_sounds.go:16]: just unexport the type and make it an internal matter, no need to expose this outside

**Comment by sebm253** [comment]
closing this for now

**Pull request closed without merging**